ngraph/core/src/op/bucketize.cpp

```diff
@@ -44,15 +44,28 @@ void op::v3::Bucketize::validate_and_infer_types()
     const PartialShape& data_pshape = get_input_partial_shape(0);
     const PartialShape& buckets_pshape = get_input_partial_shape(1);
 
+    const auto data_et = get_input_element_type(0);
+    const auto buckets_et = get_input_element_type(1);
+
+    NODE_VALIDATION_CHECK(this,
+                          data_et.is_real() || data_et.is_integral_number(),
+                          "Data input type must be numeric. Got: ",
+                          data_et);
+
+    NODE_VALIDATION_CHECK(this,
+                          buckets_et.is_real() || buckets_et.is_integral_number(),
+                          "Buckets input type must be numeric. Got: ",
+                          buckets_et);
```

Comment:
As I see in the reference implementation `buckets_et` and `data_et` should be the same (`const T* data,  const T* buckets,`). Should we add this check?

Reply:
I have addressed this differently. Based on spec, bucketize inputs maybe of different type. Additionally, this op should behave as in tf [implementation](https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/bucketize). data inputs may be int or floats, while buckets input are float.
[CPU plugin implementation](https://github.com/ggalieroc/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/bucketize.cpp#L81) is aligned with tf spec for data input types, while for buckets input types is less restrictive, allowing both int and float types. The latest changes are aligned with this behavior of CPU plugin.